x-pack/plugins/security_solution/public/management/pages/event_filters/store/middleware.ts

```diff
@@ -223,65 +223,76 @@ const checkIfEventFilterDataExist: MiddlewareActionHandler = async (
-    const query: EventFiltersServiceGetListOptions = {
-      page: pageIndex + 1,
-      perPage: pageSize,
-      sortField: 'created_at',
-      sortOrder: 'desc',
-      filter: parseQueryFilterToKQL(filter, SEARCHABLE_FIELDS) || undefined,
-    };
+  const { page_size: pageSize, page_index: pageIndex, filter } = getCurrentLocation(state);
+  const query: EventFiltersServiceGetListOptions = {
+    page: pageIndex + 1,
+    perPage: pageSize,
+    sortField: 'created_at',
+    sortOrder: 'desc',
+    filter:
+      parseQueryFilterToKQL(queryForced !== undefined ? queryForced : filter, SEARCHABLE_FIELDS) ||
+      undefined,
+  };
 
-    try {
-      const results = await eventFiltersService.getList(query);
+  try {
+    const results = await eventFiltersService.getList(query);
 
-      dispatch({
-        type: 'eventFiltersListPageDataChanged',
-        payload: createLoadedResourceState({
-          query: { ...query, filter },
-          content: results,
-        }),
-      });
+    dispatch({
+      type: 'eventFiltersListPageDataChanged',
+      payload: createLoadedResourceState({
+        query: { ...query, filter },
+        content: results,
+      }),
+    });
 
-      // If no results were returned, then just check to make sure data actually exists for
-      // event filters. This is used to drive the UI between showing "empty state" and "no items found"
-      // messages to the user
-      if (results.total === 0) {
-        await checkIfEventFilterDataExist(store, eventFiltersService);
-      } else {
-        dispatch({
-          type: 'eventFiltersListPageDataExistsChanged',
-          payload: {
-            type: 'LoadedResourceState',
-            data: Boolean(results.total),
-          },
-        });
-      }
-    } catch (error) {
+    // If no results were returned, then just check to make sure data actually exists for
+    // event filters. This is used to drive the UI between showing "empty state" and "no items found"
+    // messages to the user
+    if (results.total === 0) {
+      await checkIfEventFilterDataExist(store, eventFiltersService);
+    } else {
       dispatch({
-        type: 'eventFiltersListPageDataChanged',
-        payload: createFailedResourceState<EventFiltersListPageData>(error.body ?? error),
+        type: 'eventFiltersListPageDataExistsChanged',
+        payload: {
+          type: 'LoadedResourceState',
+          data: Boolean(results.total),
+        },
       });
     }
+  } catch (error) {
+    dispatch({
+      type: 'eventFiltersListPageDataChanged',
+      payload: createFailedResourceState<EventFiltersListPageData>(error.body ?? error),
+    });
   }
 };
 
+const refreshListDataIfNeeded: MiddlewareActionHandler = async (store, eventFiltersService) => {
+  const { getState } = store;
+  const state = getState();
+  const isLoading = getListIsLoading(state);
+
+  if (!isLoading && listDataNeedsRefresh(state)) refreshList(store, eventFiltersService);
```

Comment:
I believe it's more readable when if statements are enclosed within `{}`s.

Reply:
Yeah, this code is outdated